README.md

```diff
@@ -1,3 +1,177 @@
-## TLSNotary Infrastructure
+## TLSNotary Deployments
 
-This repo consists of configuration files and CI/CD automations and workflows for the TLSNotary SGX based deployment in Azure cloud.
\ No newline at end of file
+This repo provides a CI/CD pipeline that automates deployment of TLSNotary backend and frontend services to Azure SGX-based virtual machines.  
+Supporting infrastructure and necessary IaC components are defined in:  
+https://github.com/privacy-scaling-explorations/devops/tree/azure/tlsnotary/terraform/azure
+
+This document focuses only on the permission model and deployment structure. Azure resource setup (e.g., networking or virtual machines) is out of scope.
+
+## Deployment Workflow
+
+The GitHub Actions workflow for deployment is currently manually triggered and assumes all required container images are pre-built.
+
+### Parameters
+
+The workflow accepts the following variables:
+
+- Resource group name
+- Environment type (`prod` or `test`)
+- Source branch
+- Deployment target (`frontend`, `backend`, or `both`)
+
+> Note: Deployments to production are only allowed from the `main` branch.
+
+### Permissions
+
+#### GitHub Runner
+
+The runner uses MS Entra federated identity to authenticate and receives short-lived credentials associated with a Service Principal that has:
+
+- **Reader** role on the target resource group (to query resources and locate the appropriate VMs using tags)
+- **Azure Run Command** permission to execute deployment logic remotely on the VMs
+
+#### Backend VMs
+
+Backend VMs must have a system-assigned managed identity with read access to the Azure Key Vault. Access is granted using **RBAC-based roles** (e.g., Key Vault Reader).
+
+## Deployment Instructions
+
+### 1. Update Docker Compose
+
+Add new notary services using the following pattern:
+
+```yaml
+alpha9-sgx:
+  container_name: alpha9-sgx
+  labels:
+    - "traefik.enable=true"
+    - "traefik.http.middlewares.stripalpha9-sgx.stripprefix.prefixes=/alpha9-sgx"
+    - "traefik.http.routers.alpha9-sgx.middlewares=stripalpha9-sgx"
+  image: ghcr.io/tlsnotary/tlsn/notary-server-sgx:v0.1.0-alpha.9
+  restart: unless-stopped
+  devices:
+    - /dev/sgx_enclave
+    - /dev/sgx_provision
+  volumes:
+    - /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket
+  ports:
+    - "4109:7047"
+  entrypoint: [ "gramine-sgx", "notary-server" ]
+```
+
+- Each new service must define a `traefik` HTTP router and a strip-prefix middleware to route requests correctly.
+- Also update the landing page in `proxy/index.html` to reflect the new service route.
+
+### 2. Provision Secrets in Azure Key Vault
+
+Use the `upload-secrets.sh` script located in this repo.
+
+#### Requirements
+
+- Tools: `yq`, `az-cli`
+- The user or automation agent must have permission to write to the Azure Key Vault.
+
+#### Directory Structure
+
+The script expects the following structure relative to your current working directory:
+
+```
+.
+├── alpha10
+│   └── fixture
```

Comment:
The fixture is still optional?

Reply:
Yes. If we upload secrets to the key vault with the procedure shown in README, the deployment workflow will pull them (it will know what to look for based on the compose service names). If no secrets exist in key vault, it will skip. So deployment action covers all cases regardless of if a fixture directory exists.